### PR TITLE
Add tamed turtle

### PR DESCRIPTION
**Pull request by zyxkad**
**PLEASE READ THE [GUIDELINES](https://github.com/SirEndii/AdvancedPeripherals/blob/dev/1.20.1/CONTRIBUTING.md) BEFORE MAKING A CONTRIBUTION**

* **Please check if the PR fulfills these requirements**
- [x] The commit message are well described
- [ ] Docs have been added / updated (for features or maybe bugs which were noted). If not, please update the needed documentation [here](https://github.com/SirEndii/Advanced-Peripherals-Documentation/pulls). This is not mandatory
- [x] All changes have fully been tested

* **What kind of change does this PR introduce?** (Bug fix, feature, ...)
   Feature

* **What is the current behavior?** (You can also link to an open issue here)

* **What is the new behavior (if this is a feature change)?**
  - Saddle turtle can capture player by invoking `capture` peripheral method, and `release` for dismount
  - It also can use `getRider` to get the information of the captured entity
  - When player is captured, press the inventory key (default `E`) can open the turtle's GUI
  - Updated all deprecated functions

* **Does this PR introduce a breaking change?** (What changes might users need to make in their scripts due to this PR?)

* **Other information**:

### CODE REVIEW

**Comment by zyxkad** [comment]
HUD stuff to show turtle fuel require Mixins and a read lot of minecraft asm, I don't think if I can done that recently.

If I got the HUD bar working, I'll make another PR

**Comment by zyxkad** [comment]
As well as the saddle turtle's model need better saddle drawing, rather than just show the saddle on the side ...

**Comment by zyxkad** [comment]
But anyways, forge already did that mixin for you, so the HUD is done 💦

File: src/main/java/de/srendi/advancedperipherals/common/setup/APEntities.java [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+            .clientTrackingRange(4)
+            .updateInterval(4)
+            .fireImmune()
+            .build("turtle_seat"));
+
+    public static void register() {
+    }
+
+    @SubscribeEvent
+    public static void livingRender(EntityRenderersEvent.RegisterRenderers event) {
```
Comment (SirEndii) [inline src/main/java/de/srendi/advancedperipherals/common/setup/APEntities.java:28]: Do we maybe want to register this in our dedicated ClientRegistry class?

**Review by SirEndii — CHANGES_REQUESTED** [review]

**Comment by zyxkad** [comment]
@SirEndii This is ready for review again

**Comment by SirEndii** [comment]
gonna review that tomorrow
and I need to fix the actions stuff with prs

File: src/main/java/de/srendi/advancedperipherals/common/util/LuaConverter.java [hunk 2]
```diff
@@ -31,29 +38,64 @@
 
 public class LuaConverter {
 
-    public static Map<String, Object> entityToLua(Entity entity) {
+    public static Map<String, Object> entityToLua(Entity entity, boolean detailed) {
```
Comment (SirEndii) [inline src/main/java/de/srendi/advancedperipherals/common/util/LuaConverter.java:41]: Is it intended that here are entity related changes in the PR?

Comment (zyxkad) [reply src/main/java/de/srendi/advancedperipherals/common/util/LuaConverter.java:41]: yes cause the turtle need to get more details of its rider

**Review by SirEndii — APPROVED** [review]
LGTM

**Pull request merged**